modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/cache/ClientCacheKeyRequest.java

```diff
@@ -38,6 +45,39 @@ public abstract class ClientCacheKeyRequest extends ClientCacheDataRequest imple
         key = reader.readObjectDetached();
     }
 
+    /** {@inheritDoc} */
+    @Override public final ClientResponse process(ClientConnectionContext ctx) {
+        if (!isTransactional()) {
+            // Calculate affinity metrics.
+            DynamicCacheDescriptor desc = cacheDescriptor(ctx);
+            CacheConfiguration<?, ?> cfg = desc.cacheConfiguration();
+
+            if (cfg.getCacheMode() == CacheMode.PARTITIONED && cfg.isStatisticsEnabled()) {
+                String cacheName = desc.cacheName();
+
+                try {
+                    GridKernalContext kctx = ctx.kernalContext();
+
+                    if (F.first(kctx.affinity().mapKeyToPrimaryAndBackups(cacheName, key, null)).isLocal())
+                        kctx.clientListener().metrics().onAffinityKeyHit();
+                    else
+                        kctx.clientListener().metrics().onAffinityKeyMiss();
+                }
+                catch (Exception ignored) {
+                    // No-op.
+                }
+            }
+        }
+
+        // Process request in overriden method.
+        return process0(ctx);
+    }
+
+    /** */
+    protected ClientResponse process0(ClientConnectionContext ctx) {
+        return super.process(ctx);
```

Comment:
Do we really should call `super.process` here? Looks very suspicious and fragile for me.

Reply:
This logic was inherited from the previous implementation (common response was created by `super.process()` method), and such approach is used by some other requests. But I can change it and create response explicitly, there are only two usages of `super.process0` method now.